File: scripts/read_pdfs.py
import PyPDF2
import os

files = [
    'public/notes/A Level/Business/theory.pdf',
    'public/notes/A Level/Business/definitions.pdf',
    'public/notes/A Level/Business/formulae.pdf'
]

for file_path in files:
    if os.path.exists(file_path):
        print(f"\n\n=== START: {file_path} ===\n")
        with open(file_path, 'rb') as f:
            reader = PyPDF2.PdfReader(f)
            for page in reader.pages:
                print(page.extract_text())
        print(f"\n=== END: {file_path} ===\n")
    else:
        print(f"File not found: {file_path}")

File: scripts/read_pdfs.js
const fs = require('fs');
const pdf = require('pdf-parse');

async function extract() {
  const files = [
    'public/notes/A Level/Business/theory.pdf',
    'public/notes/A Level/Business/definitions.pdf',
    'public/notes/A Level/Business/formulae.pdf'
  ];

  for (const file of files) {
    if (fs.existsSync(file)) {
      const dataBuffer = fs.readFileSync(file);
      const data = await pdf(dataBuffer);
      console.log(`\n\n=== START: ${file} ===\n`);
      console.log(data.text.trim());
      console.log(`\n=== END: ${file} ===\n`);
    } else {
        console.log(`File not found: ${file}`);
    }
  }
}

extract().catch(console.error);
